src/models/multi_modal_vision_model.cpp

```diff
@@ -185,22 +92,17 @@ RoamingArray<float> EmbeddingState::Run(int current_length, RoamingArray<int32_t
   return MakeDummy();
 }
 
-VisionState::VisionState(const MultiModalVisionModel& model, const GeneratorParams& params)
+VisionState::VisionState(const MultiModalVisionModel& model, const GeneratorParams& params, const int64_t num_image_tokens)
     : State{params, model},
-      model_{model} {
+      model_{model},
+      num_image_tokens_{num_image_tokens} {
   extra_inputs_.Add();
-  num_image_tokens_ = static_cast<int32_t>(GetNumImageTokens(params_->extra_inputs, model_.config_->model.vision.inputs.image_sizes));
-  if (num_image_tokens_ > 0) {
-    visual_features_ = GetVisualFeatures(*model_.allocator_device_, *model_.session_info_,
-                                         model_.config_->model.vision.outputs.visual_features,
-                                         params_->hidden_size, num_image_tokens_);
-    output_names_.push_back(model_.config_->model.vision.outputs.visual_features.c_str());
-    outputs_.push_back(visual_features_.get());
-  }
+  image_features_.Add();
 }
 
 RoamingArray<float> VisionState::Run(int current_length, RoamingArray<int32_t> next_tokens, RoamingArray<int32_t> next_indices) {
-  State::Run(*model_.vision_session_, *model_.run_options_, 1);
+  int num_images = static_cast<int>(inputs_[0]->GetTensorTypeAndShapeInfo()->GetShape()[0]);
```

Comment:
Is the batch size dimension = num_images?
Is it possible to run this batched? Multiple images in a batch?

Reply:
The two dimensions are different. For example, you can pass 2 images and 1 prompt. Then `num_images = 2` in the vision component and `batch_size = 1` in the text component.

Reply:
So the vision component cannot be run with a batch_size > 1 for now, right?

Reply:
The `num_images` is the batch size for the vision component and the `batch_size` is the batch size for the text component. With these changes, you can run this model batched (i.e. multiple images in a batch) as well. The processor just has to prepare the inputs correctly.